{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib notebook\n",
    "\n",
    "import os\n",
    "import glob\n",
    "import zipfile\n",
    "import shutil\n",
    "\n",
    "import pandas\n",
    "\n",
    "import gp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "source_dir = \"/Users/dlahr/temp/H202SC20040591_dge/analysis/\"\n",
    "experiment_id = \"H202SC20040591\"\n",
    "\n",
    "dge_stats_for_rnk_list = [\"logFC\", \"t\"]\n",
    "\n",
    "num_permutations = 1000\n",
    "\n",
    "job_memory = \"4 Gb\"\n",
    "\n",
    "# guidance from GenePattern forum:\n",
    "# gseapreranked_job_spec.set_parameter(\"job.memory\", \"2 Gb\")\n",
    "# gseapreranked_job_spec.set_parameter(\"job.queue\", \"gp-cloud-default\")\n",
    "# gseapreranked_job_spec.set_parameter(\"job.cpuCount\", \"1\")\n",
    "# gseapreranked_job_spec.set_parameter(\"job.walltime\", \"02:00:00\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# notes from running if any\n",
    "Job 257826 failed for some reason, restarted manually on server, will download / unzip manually as well when ready"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# prepare directories for output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gsea_dir: /Users/dlahr/temp/H202SC20040591_dge/analysis/gsea\n",
      "rnk_dir: /Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk\n"
     ]
    }
   ],
   "source": [
    "gsea_dir = os.path.join(source_dir, \"gsea\")\n",
    "print(\"gsea_dir:\", gsea_dir)\n",
    "if os.path.exists(gsea_dir):\n",
    "    shutil.rmtree(gsea_dir)\n",
    "os.mkdir(gsea_dir)\n",
    "\n",
    "rnk_dir = os.path.join(gsea_dir, \"rnk\")\n",
    "print(\"rnk_dir:\", rnk_dir)\n",
    "os.mkdir(rnk_dir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# build rnk files from DGE data files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_DGE_r30500x10.txt\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_DGE_r30500x10.txt\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_DGE_r30500x10.txt\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_DGE_r30500x10.txt\n"
     ]
    }
   ],
   "source": [
    "dge_file_list = glob.glob(\n",
    "    os.path.join(source_dir, \"dge_data\", experiment_id + \"_*_DGE_r*.txt\")\n",
    ")\n",
    "dge_file_list.sort()\n",
    "print(len(dge_file_list))\n",
    "print(\"\\n\".join(dge_file_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dge_file: /Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_DGE_r30500x10.txt\n",
      "(30500, 11)\n",
      "H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_DGE_r30500x10\n",
      "base_output_filename: H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>logFC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-160E2.16</td>\n",
       "      <td>-3.495766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP1-99E18.2</td>\n",
       "      <td>-3.032918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RP11-386M24.9</td>\n",
       "      <td>-2.944619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RP11-468E2.10</td>\n",
       "      <td>-2.915015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEFF2</td>\n",
       "      <td>-2.797327</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #gene_symbol     logFC\n",
       "0  RP11-160E2.16 -3.495766\n",
       "1    RP1-99E18.2 -3.032918\n",
       "2  RP11-386M24.9 -2.944619\n",
       "3  RP11-468E2.10 -2.915015\n",
       "4         TMEFF2 -2.797327"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.rnk\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>t</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-160E2.16</td>\n",
       "      <td>-4.864835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP1-99E18.2</td>\n",
       "      <td>-4.682609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>RP11-386M24.9</td>\n",
       "      <td>-3.918192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>RP11-468E2.10</td>\n",
       "      <td>-3.826496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TMEFF2</td>\n",
       "      <td>-3.284992</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #gene_symbol         t\n",
       "0  RP11-160E2.16 -4.864835\n",
       "1    RP1-99E18.2 -4.682609\n",
       "2  RP11-386M24.9 -3.918192\n",
       "3  RP11-468E2.10 -3.826496\n",
       "4         TMEFF2 -3.284992"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.rnk\n",
      "\n",
      "dge_file: /Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_DGE_r30500x10.txt\n",
      "(30500, 11)\n",
      "H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_DGE_r30500x10\n",
      "base_output_filename: H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>logFC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYH4</td>\n",
       "      <td>-4.282970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MTND6P22</td>\n",
       "      <td>-3.887673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYH2</td>\n",
       "      <td>-3.757082</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MYH1</td>\n",
       "      <td>-3.474597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RP11-70P17.1</td>\n",
       "      <td>-3.457608</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #gene_symbol     logFC\n",
       "0          MYH4 -4.282970\n",
       "1      MTND6P22 -3.887673\n",
       "2          MYH2 -3.757082\n",
       "3          MYH1 -3.474597\n",
       "4  RP11-70P17.1 -3.457608"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.rnk\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>t</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MYH4</td>\n",
       "      <td>-1.369989</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MTND6P22</td>\n",
       "      <td>-5.600535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MYH2</td>\n",
       "      <td>-1.424892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MYH1</td>\n",
       "      <td>-1.279057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>RP11-70P17.1</td>\n",
       "      <td>-5.355376</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #gene_symbol         t\n",
       "0          MYH4 -1.369989\n",
       "1      MTND6P22 -5.600535\n",
       "2          MYH2 -1.424892\n",
       "3          MYH1 -1.279057\n",
       "4  RP11-70P17.1 -5.355376"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.rnk\n",
      "\n",
      "dge_file: /Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_DGE_r30500x10.txt\n",
      "(30500, 11)\n",
      "H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_DGE_r30500x10\n",
      "base_output_filename: H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>logFC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-67H2.1</td>\n",
       "      <td>-3.333734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-148B3.1</td>\n",
       "      <td>-3.330712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZDHHC20-IT1</td>\n",
       "      <td>-3.250020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>STRCP1</td>\n",
       "      <td>-3.229273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR51B5</td>\n",
       "      <td>-3.157891</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #gene_symbol     logFC\n",
       "0   RP11-67H2.1 -3.333734\n",
       "1  RP11-148B3.1 -3.330712\n",
       "2   ZDHHC20-IT1 -3.250020\n",
       "3        STRCP1 -3.229273\n",
       "4        OR51B5 -3.157891"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.rnk\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>t</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-67H2.1</td>\n",
       "      <td>-5.038409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-148B3.1</td>\n",
       "      <td>-4.730135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZDHHC20-IT1</td>\n",
       "      <td>-4.327745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>STRCP1</td>\n",
       "      <td>-4.738674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OR51B5</td>\n",
       "      <td>-5.495765</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   #gene_symbol         t\n",
       "0   RP11-67H2.1 -5.038409\n",
       "1  RP11-148B3.1 -4.730135\n",
       "2   ZDHHC20-IT1 -4.327745\n",
       "3        STRCP1 -4.738674\n",
       "4        OR51B5 -5.495765"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.rnk\n",
      "\n",
      "dge_file: /Users/dlahr/temp/H202SC20040591_dge/analysis/dge_data/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_DGE_r30500x10.txt\n",
      "(30500, 11)\n",
      "H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_DGE_r30500x10\n",
      "base_output_filename: H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>logFC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-160E2.17</td>\n",
       "      <td>-5.171117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-160E2.16</td>\n",
       "      <td>-5.075931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COL10A1</td>\n",
       "      <td>-4.861402</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMHR2</td>\n",
       "      <td>-4.534430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AC007952.4</td>\n",
       "      <td>-4.166896</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #gene_symbol     logFC\n",
       "0  RP11-160E2.17 -5.171117\n",
       "1  RP11-160E2.16 -5.075931\n",
       "2        COL10A1 -4.861402\n",
       "3          AMHR2 -4.534430\n",
       "4     AC007952.4 -4.166896"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.rnk\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#gene_symbol</th>\n",
       "      <th>t</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>RP11-160E2.17</td>\n",
       "      <td>-9.588262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RP11-160E2.16</td>\n",
       "      <td>-8.472290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COL10A1</td>\n",
       "      <td>-6.522934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMHR2</td>\n",
       "      <td>-8.484385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AC007952.4</td>\n",
       "      <td>-7.973930</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    #gene_symbol         t\n",
       "0  RP11-160E2.17 -9.588262\n",
       "1  RP11-160E2.16 -8.472290\n",
       "2        COL10A1 -6.522934\n",
       "3          AMHR2 -8.484385\n",
       "4     AC007952.4 -7.973930"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2.rnk\n",
      "\n",
      "8\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.rnk\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/rnk/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2.rnk\n"
     ]
    }
   ],
   "source": [
    "input_rnk_files_list = []\n",
    "\n",
    "for dge_file in dge_file_list[:]:\n",
    "    print(\"dge_file:\", dge_file)\n",
    "    dge_df = pandas.read_csv(dge_file, sep=\"\\t\")\n",
    "    print(dge_df.shape)\n",
    "    \n",
    "    base_dge_filename = os.path.splitext(os.path.basename(dge_file))[0]\n",
    "    print(base_dge_filename)\n",
    "    \n",
    "    base_output_filename = \"_\".join(base_dge_filename.split(\"_\")[:-2])\n",
    "    print(\"base_output_filename:\", base_output_filename)\n",
    "    \n",
    "    for dge_stat_for_rnk in dge_stats_for_rnk_list:\n",
    "        rnk_df = dge_df.loc[~pandas.isnull(dge_df.gene_symbol), \n",
    "                            [\"gene_symbol\", dge_stat_for_rnk]].copy()\n",
    "        new_cols = list(rnk_df.columns)\n",
    "        new_cols[0] = \"#\" + new_cols[0]\n",
    "        rnk_df.columns = new_cols\n",
    "        display(rnk_df.head())\n",
    "        \n",
    "        output_rnk_file = base_output_filename + \"_{dge_stat}_r{rows}x{cols}.rnk\".format(\n",
    "            dge_stat=dge_stat_for_rnk, rows=rnk_df.shape[0], cols=rnk_df.shape[1]\n",
    "        )\n",
    "        print(output_rnk_file)\n",
    "        \n",
    "        output_filepath = os.path.join(rnk_dir, output_rnk_file)\n",
    "        input_rnk_files_list.append(output_filepath)\n",
    "        rnk_df.to_csv(output_filepath, sep=\"\\t\", index=False)\n",
    "    \n",
    "    print()\n",
    "\n",
    "print(len(input_rnk_files_list))\n",
    "print(\"\\n\".join(input_rnk_files_list))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# run GSEA using rnk files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2\n",
      "[('all', 7), ('just_hallmarks', 1)]\n"
     ]
    }
   ],
   "source": [
    "reference_genesets = [\n",
    "    (\"all\", {\"c1.all.v7.1.symbols.gmt [Positional]\", \"c2.all.v7.1.symbols.gmt [Curated]\", \"c3.all.v7.1.symbols.gmt [Motif]\",\n",
    "     \"c5.all.v7.1.symbols.gmt [Gene Ontology]\", \"c6.all.v7.1.symbols.gmt [Oncogenic Signatures]\",\n",
    "     \"c7.all.v7.1.symbols.gmt [Immunologic signatures]\", \"h.all.v7.1.symbols.gmt [Hallmarks]\"}),\n",
    "    (\"just_hallmarks\", {\"h.all.v7.1.symbols.gmt [Hallmarks]\"})\n",
    "]\n",
    "print(len(reference_genesets))\n",
    "print([(k,len(v)) for k,v in reference_genesets])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gp_username = \"dllahr\"\n",
    "gp_password = \"yQJ7a6m^mBE07#845@Ll#AJq!\"\n",
    "gp_url = \"https://cloud.genepattern.org/gp\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a GenePattern server proxy instance\n",
    "gpserver = gp.GPServer(gp_url, gp_username, gp_password)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run5687728369687963663.tmp/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run7751554138625777331.tmp/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run1005952227850521048.tmp/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run7904784910493905373.tmp/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run6076331056623571247.tmp/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run3832421169152986486.tmp/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run5196629571769072772.tmp/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.rnk\n",
      "H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2.rnk\n",
      "https://cloud.genepattern.org/gp/users/dllahr/tmp/run2031244483517157540.tmp/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2.rnk\n"
     ]
    }
   ],
   "source": [
    "# upload input files\n",
    "input_gp_files_list = []\n",
    "\n",
    "for input_file in input_rnk_files_list:\n",
    "    uploaded_file_name = os.path.basename(input_file)\n",
    "    print(uploaded_file_name)\n",
    "\n",
    "    input_gp_file = gpserver.upload_file(uploaded_file_name, input_file)\n",
    "#     print(dir(input_gp_file))\n",
    "    print(input_gp_file.get_url())\n",
    "    input_gp_files_list.append(input_gp_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get the list of tasks\n",
    "task_list = gpserver.get_task_list()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(\"\\n\".join([x.get_name() for x in task_list]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4\n",
      "GSEA\n",
      "GSEALeadingEdgeViewer\n",
      "GSEAPreranked\n",
      "ssGSEA\n"
     ]
    }
   ],
   "source": [
    "gsea_task_list = [x for x in task_list if \"gsea\" in x.get_name().lower()]\n",
    "print(len(gsea_task_list))\n",
    "print(\"\\n\".join([x.get_name() for x in gsea_task_list]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_params_loaded', 'description', 'documentation', 'dto', 'get_description', 'get_lsid', 'get_name', 'get_parameters', 'get_version', 'job', 'job_number', 'job_spec', 'json', 'lsid', 'make_job_spec', 'name', 'param_load', 'params', 'server_data', 'submit_json', 'uri', 'version']\n",
      "GSEAPreranked\n",
      "urn:lsid:broad.mit.edu:cancer.software.genepattern.module.analysis:00317:7.0.4\n"
     ]
    }
   ],
   "source": [
    "t = [x for x in gsea_task_list if x.get_name() == \"GSEAPreranked\"]\n",
    "assert len(t) == 1\n",
    "gsea_preranked_task = t[0]\n",
    "print(dir(gsea_preranked_task))\n",
    "print(gsea_preranked_task.name)\n",
    "print(gsea_preranked_task.lsid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_params_loaded', 'description', 'documentation', 'dto', 'get_description', 'get_lsid', 'get_name', 'get_parameters', 'get_version', 'job', 'job_number', 'job_spec', 'json', 'lsid', 'make_job_spec', 'name', 'param_load', 'params', 'server_data', 'submit_json', 'uri', 'version']\n"
     ]
    }
   ],
   "source": [
    "gsea_preranked_module = gp.GPTask(gpserver, gsea_preranked_task.lsid)\n",
    "print(dir(gsea_preranked_module))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "gsea_preranked_module.param_load()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ranked.list\n",
      "type: File\n",
      "Rank ordered gene list - .rnk\n",
      "default_value: None\n",
      "is_optional: False\n",
      "\n",
      "gene.sets.database\n",
      "type: File\n",
      "Gene sets database from GSEA website.\n",
      "default_value: None\n",
      "is_optional: False\n",
      "\n",
      "number.of.permutations\n",
      "type: String\n",
      "Number of permutations to perform\n",
      "default_value: 1000\n",
      "is_optional: False\n",
      "\n",
      "collapse.dataset\n",
      "type: String\n",
      "Select whether to collapse each probe set in the expression dataset into a single vector for the gene, which gets identified by its gene symbol.  It is also possible to remap symbols from one namespace to another without collapsing (an error will occur if multiple source genes map to a single destination gene).  No_Collapse will use the dataset as-is.\n",
      "default_value: Remap_Only\n",
      "is_optional: False\n",
      "\n",
      "chip.platform.file\n",
      "type: File\n",
      "DNA Chip (array) annotation file from GSEA website.  Upload your own chip file if the one corresponding to your DNA Microarray platform is not listed in the drop-down menu.  A chip file is only required if collapse dataset is set to true.\n",
      "default_value: None\n",
      "is_optional: False\n",
      "\n",
      "scoring.scheme\n",
      "type: String\n",
      "The statistic used to score hits (gene set members) and misses (non-members)\n",
      "default_value: weighted\n",
      "is_optional: False\n",
      "\n",
      "max.gene.set.size\n",
      "type: String\n",
      "Gene sets larger than this are excluded from the analysis\n",
      "default_value: 500\n",
      "is_optional: False\n",
      "\n",
      "min.gene.set.size\n",
      "type: String\n",
      "Gene sets smaller than this are excluded from the analysis\n",
      "default_value: 15\n",
      "is_optional: False\n",
      "\n",
      "collapsing.mode.for.probe.sets.with.more.than.one.match\n",
      "type: String\n",
      "Collapsing mode.  This has no effect when remapping symbols.  For custom ranking metrics, be very cautious when selecting any of these modes to be sure it is compatible with your metric. \n",
      "default_value: Max_probe\n",
      "is_optional: False\n",
      "\n",
      "normalization.mode\n",
      "type: String\n",
      "Normalization mode\n",
      "default_value: meandiv\n",
      "is_optional: False\n",
      "\n",
      "omit.features.with.no.symbol.match\n",
      "type: String\n",
      "If there is no known gene symbol match for a probe set, omit it from the collapsed dataset\n",
      "default_value: true\n",
      "is_optional: False\n",
      "\n",
      "make.detailed.gene.set.report\n",
      "type: String\n",
      "Create detailed gene set report (heat map, mountain plot, etc.) for every enriched gene set\n",
      "default_value: true\n",
      "is_optional: False\n",
      "\n",
      "num.top.sets\n",
      "type: String\n",
      "Plot GSEA mountain and related plots for the top sets of each phenotype\n",
      "default_value: 20\n",
      "is_optional: False\n",
      "\n",
      "random.seed\n",
      "type: String\n",
      "Seed to use for generating random permutations.\n",
      "default_value: timestamp\n",
      "is_optional: False\n",
      "\n",
      "create.svgs\n",
      "type: String\n",
      "Whether to create SVG images (compressed) along with PNGs\n",
      "default_value: false\n",
      "is_optional: False\n",
      "\n",
      "output.file.name\n",
      "type: String\n",
      "Name of the output file.  Note that this only applies if create.zip=true.\n",
      "default_value: <ranked.list_basename>.zip\n",
      "is_optional: False\n",
      "\n",
      "selected.gene.sets\n",
      "type: String\n",
      "Semicolon-separated list of gene sets from the provided gene sets database files (GMT/GMX/GRP).  If you are using multiple files then you *must* prefix each selected gene set with its file name followed by '#' (like \"my_file1.gmt#selected_gene_set1,my_file2.gmt#selected_gene_set2\").  With a single file only the names are necessary.  Leave this blank to select all gene sets.\n",
      "default_value: None\n",
      "is_optional: False\n",
      "\n",
      "alt.delim\n",
      "type: String\n",
      "Optional alternate delimiter character for gene set names instead of comma for use with selected.gene.sets.  If used, a semicolon is recommended.\n",
      "default_value: None\n",
      "is_optional: False\n",
      "\n",
      "create.zip\n",
      "type: String\n",
      "Create a ZIP bundle of the output files.\n",
      "default_value: true\n",
      "is_optional: False\n",
      "\n",
      "dev.mode\n",
      "type: String\n",
      "Enable developer mode.\n",
      "default_value: false\n",
      "is_optional: False\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Get the list of GPTaskParam objects\n",
    "params_list = gsea_preranked_module.get_parameters()\n",
    "\n",
    "for param in params_list:              # Loop through each parameter\n",
    "    print( param.get_name() )          # Print the parameter's name\n",
    "    print( \"type:\", param.get_type() )          # Print the parameter's type (text, number, file, etc.)\n",
    "    print( param.get_description() )   # Print the parameter's description\n",
    "    print( \"default_value:\", param.get_default_value() ) # Print the parameter's default value\n",
    "    print( \"is_optional:\", param.is_optional() )       # Print whether the parameter is optional\n",
    "    print( '' )                        # Leave a blank line between printed parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene.sets.database\n",
      "c1.all.v7.1.symbols.gmt [Positional] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c1.all.v7.1.symbols.gmt\n",
      "c2.all.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.all.v7.1.symbols.gmt\n",
      "c2.cgp.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cgp.v7.1.symbols.gmt\n",
      "c2.cp.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cp.v7.1.symbols.gmt\n",
      "c2.cp.biocarta.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cp.biocarta.v7.1.symbols.gmt\n",
      "c2.cp.kegg.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cp.kegg.v7.1.symbols.gmt\n",
      "c2.cp.pid.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cp.pid.v7.1.symbols.gmt\n",
      "c2.cp.reactome.v7.1.symbols.gmt [Curated] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.cp.reactome.v7.1.symbols.gmt\n",
      "c3.all.v7.1.symbols.gmt [Motif] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c3.all.v7.1.symbols.gmt\n",
      "c3.mir.v7.1.symbols.gmt [Motif] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c3.mir.v7.1.symbols.gmt\n",
      "c3.tft.v7.1.symbols.gmt [Motif] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c3.tft.v7.1.symbols.gmt\n",
      "c4.all.v7.1.symbols.gmt [Computational] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c4.all.v7.1.symbols.gmt\n",
      "c4.cgn.v7.1.symbols.gmt [Computational] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c4.cgn.v7.1.symbols.gmt\n",
      "c4.cm.v7.1.symbols.gmt [Computational] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c4.cm.v7.1.symbols.gmt\n",
      "c5.all.v7.1.symbols.gmt [Gene Ontology] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.all.v7.1.symbols.gmt\n",
      "c5.bp.v7.1.symbols.gmt [Gene Ontology] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.bp.v7.1.symbols.gmt\n",
      "c5.cc.v7.1.symbols.gmt [Gene Ontology] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.cc.v7.1.symbols.gmt\n",
      "c5.mf.v7.1.symbols.gmt [Gene Ontology] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.mf.v7.1.symbols.gmt\n",
      "c6.all.v7.1.symbols.gmt [Oncogenic Signatures] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c6.all.v7.1.symbols.gmt\n",
      "c7.all.v7.1.symbols.gmt [Immunologic signatures] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c7.all.v7.1.symbols.gmt\n",
      "h.all.v7.1.symbols.gmt [Hallmarks] = ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/h.all.v7.1.symbols.gmt\n",
      "ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c1.all.v7.1.symbols.gmt\n",
      "\n",
      "collapse.dataset\n",
      "No_Collapse = No_Collapse\n",
      "Collapse = Collapse\n",
      "Remap_Only = Remap_Only\n",
      "Remap_Only\n",
      "\n",
      "chip.platform.file\n",
      " = \n",
      "Human_AFFY_HC_G110_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HC_G110_MSigDB.v7.0.chip\n",
      "Human_AFFY_HC_G110_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HC_G110_MSigDB.v7.1.chip\n",
      "Human_AFFY_HG_Focus_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_Focus_MSigDB.v7.0.chip\n",
      "Human_AFFY_HG_Focus_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_Focus_MSigDB.v7.1.chip\n",
      "Human_AFFY_HG_U133_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_U133_MSigDB.v7.0.chip\n",
      "Human_AFFY_HG_U133_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_U133_MSigDB.v7.1.chip\n",
      "Human_AFFY_HG_U95_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_U95_MSigDB.v7.0.chip\n",
      "Human_AFFY_HG_U95_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HG_U95_MSigDB.v7.1.chip\n",
      "Human_AFFY_HTA_2_0_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HTA_2_0_MSigDB.v7.0.chip\n",
      "Human_AFFY_HTA_2_0_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HTA_2_0_MSigDB.v7.1.chip\n",
      "Human_AFFY_HuGene_1_0_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGene_1_0_st_v1_MSigDB.v7.0.chip\n",
      "Human_AFFY_HuGene_1_0_st_v1_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGene_1_0_st_v1_MSigDB.v7.1.chip\n",
      "Human_AFFY_HuGene_2_0_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGene_2_0_st_v1_MSigDB.v7.0.chip\n",
      "Human_AFFY_HuGene_2_0_st_v1_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGene_2_0_st_v1_MSigDB.v7.1.chip\n",
      "Human_AFFY_HuGeneFL_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGeneFL_MSigDB.v7.0.chip\n",
      "Human_AFFY_HuGeneFL_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_HuGeneFL_MSigDB.v7.1.chip\n",
      "Human_AFFY_PrimeView_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_PrimeView_MSigDB.v7.0.chip\n",
      "Human_AFFY_PrimeView_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_PrimeView_MSigDB.v7.1.chip\n",
      "Human_AFFY_U133_x3p_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_U133_x3p_MSigDB.v7.0.chip\n",
      "Human_AFFY_U133_x3p_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_AFFY_U133_x3p_MSigDB.v7.1.chip\n",
      "Human_Agilent_CGH_44b_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_CGH_44b_MSigDB.v7.0.chip\n",
      "Human_Agilent_CGH_44b_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_CGH_44b_MSigDB.v7.1.chip\n",
      "Human_Agilent_GPL6848_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_GPL6848_MSigDB.v7.0.chip\n",
      "Human_Agilent_GPL6848_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_GPL6848_MSigDB.v7.1.chip\n",
      "Human_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip\n",
      "Human_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.1.chip\n",
      "Human_Agilent_SurePrint_g3_ge_8x60k_v2_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_SurePrint_g3_ge_8x60k_v2_MSigDB.v7.0.chip\n",
      "Human_Agilent_SurePrint_g3_ge_8x60k_v2_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_SurePrint_g3_ge_8x60k_v2_MSigDB.v7.1.chip\n",
      "Human_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip\n",
      "Human_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.1.chip\n",
      "Human_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.0.chip\n",
      "Human_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.1.chip\n",
      "Human_CodeLink_Array_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_CodeLink_Array_MSigDB.v7.0.chip\n",
      "Human_Codelink_Array_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Codelink_Array_MSigDB.v7.1.chip\n",
      "Human_ENSEMBL_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_ENSEMBL_Gene_ID_MSigDB.v7.0.chip\n",
      "Human_ENSEMBL_Gene_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_ENSEMBL_Gene_MSigDB.v7.1.chip\n",
      "Human_HGNC_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_HGNC_ID_MSigDB.v7.0.chip\n",
      "Human_HGNC_ID_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_HGNC_ID_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanHT_12_v3_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanHT_12_v3_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanHT_12_v3_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanHT_12_v3_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanHT_12_v4_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanHT_12_v4_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanHT_12_v4_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanHT_12_v4_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanRef_8_v3_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanRef_8_v3_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanRef_8_v3_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanRef_8_v3_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanWG_6_v1_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v1_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanWG_6_v1_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v1_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanWG_6_v2_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v2_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanWG_6_v2_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v2_MSigDB.v7.1.chip\n",
      "Human_Illumina_HumanWG_6_v3_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v3_BeadChip_MSigDB.v7.0.chip\n",
      "Human_Illumina_HumanWG_6_v3_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Illumina_HumanWG_6_v3_MSigDB.v7.1.chip\n",
      "Human_NCBI_Entrez_Gene_ID_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_NCBI_Entrez_Gene_ID_MSigDB.v7.1.chip\n",
      "Human_NCBI_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_NCBI_Gene_ID_MSigDB.v7.0.chip\n",
      "Human_Phalanx_OneArray_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Phalanx_OneArray_MSigDB.v7.0.chip\n",
      "Human_Phalanx_OneArray_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Phalanx_OneArray_MSigDB.v7.1.chip\n",
      "Human_RefSeq_Accession_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_RefSeq_Accession_MSigDB.v7.1.chip\n",
      "Human_RefSeq_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_RefSeq_ID_MSigDB.v7.0.chip\n",
      "Human_Symbol_with_Remapping_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Symbol_with_Remapping_MSigDB.v7.0.chip\n",
      "Human_Symbol_with_Remapping_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_Symbol_with_Remapping_MSigDB.v7.1.chip\n",
      "Human_UniProt_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_Gene_ID_MSigDB.v7.0.chip\n",
      "Human_UniProt_Gene_ID_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_Gene_ID_MSigDB.v7.1.chip\n",
      "Human_UniProt_IDs_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_IDs_MSigDB.v7.0.chip\n",
      "Human_UniProt_IDs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_IDs_MSigDB.v7.1.chip\n",
      "Human_UniProt_SwissProt_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_SwissProt_MSigDB.v7.0.chip\n",
      "Human_UniProt_SwissProt_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_SwissProt_MSigDB.v7.1.chip\n",
      "Human_UniProt_TrEMBL_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_TrEMBL_MSigDB.v7.0.chip\n",
      "Human_UniProt_TrEMBL_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Human_UniProt_TrEMBL_MSigDB.v7.1.chip\n",
      "Mouse_AFFY_MG_U74_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MG_U74_MSigDB.v7.0.chip\n",
      "Mouse_AFFY_MG_U74_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MG_U74_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_AFFY_MoGene_1_0_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MoGene_1_0_st_v1_MSigDB.v7.0.chip\n",
      "Mouse_AFFY_MoGene_1_0_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MoGene_1_0_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_AFFY_MoGene_2_1_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MoGene_2_1_st_v1_MSigDB.v7.0.chip\n",
      "Mouse_AFFY_MoGene_2_1_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_MoGene_2_1_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_AFFY_Mouse430_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_Mouse430_MSigDB.v7.0.chip\n",
      "Mouse_AFFY_Mouse430_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_Mouse430_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_AFFY_Mu11K_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_Mu11K_MSigDB.v7.0.chip\n",
      "Mouse_AFFY_Mu11K_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_AFFY_Mu11K_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip\n",
      "Mouse_Agilent_SurePrint_g3_ge_8x60k_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_SurePrint_g3_ge_8x60k_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip\n",
      "Mouse_Agilent_WholeGenome_4x44k_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_WholeGenome_4x44k_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_WholeGenome_4x44k_v2_MSigDB.v7.0.chip\n",
      "Mouse_Agilent_WholeGenome_4x44k_v2_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Agilent_WholeGenome_4x44k_v2_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Codelink_Array_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Codelink_Array_MSigDB.v7.0.chip\n",
      "Mouse_Codelink_Array_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Codelink_Array_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_ENSEMBL_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_ENSEMBL_Gene_ID_MSigDB.v7.0.chip\n",
      "Mouse_ENSEMBL_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_ENSEMBL_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Gene_Symbol_Remapping_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Gene_Symbol_Remapping_MSigDB.v7.0.chip\n",
      "Mouse_Gene_Symbol_Remapping_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Gene_Symbol_Remapping_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Illumina_MouseRef_8_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseRef_8_BeadChip_MSigDB.v7.0.chip\n",
      "Mouse_Illumina_MouseRef_8_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseRef_8_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Illumina_MouseWG_6_v1_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseWG_6_v1_BeadChip_MSigDB.v7.0.chip\n",
      "Mouse_Illumina_MouseWG_6_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseWG_6_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Illumina_MouseWG_6_v2_BeadChip_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseWG_6_v2_BeadChip_MSigDB.v7.0.chip\n",
      "Mouse_Illumina_MouseWG_6_v2_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Illumina_MouseWG_6_v2_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_MGI_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_MGI_ID_MSigDB.v7.0.chip\n",
      "Mouse_MGI_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_MGI_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_NCBI_Entrez_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_NCBI_Entrez_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_NCBI_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_NCBI_Gene_ID_MSigDB.v7.0.chip\n",
      "Mouse_Operon_V1_1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Operon_V1_1_MSigDB.v7.0.chip\n",
      "Mouse_Operon_V1_1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Operon_V1_1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Phalanx_OneArray_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Phalanx_OneArray_MSigDB.v7.0.chip\n",
      "Mouse_Phalanx_OneArray_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Phalanx_OneArray_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_RefSeq_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_RefSeq_ID_MSigDB.v7.0.chip\n",
      "Mouse_RefSeq_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_RefSeq_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_Seq_Accession_full_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_Seq_Accession_full_MSigDB.v7.0.chip\n",
      "Mouse_UCSC_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UCSC_ID_MSigDB.v7.0.chip\n",
      "Mouse_UCSD_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UCSD_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_UniProt_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_UniProt_IDs_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_IDs_MSigDB.v7.0.chip\n",
      "Mouse_UniProt_SwissProt_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_SwissProt_MSigDB.v7.0.chip\n",
      "Mouse_UniProt_SwissProt_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_SwissProt_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Mouse_UniProt_TrEMBL_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_TrEMBL_MSigDB.v7.0.chip\n",
      "Mouse_UniProt_TrEMBL_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Mouse_UniProt_TrEMBL_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_AFFY_230_Array_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_230_Array_MSigDB.v7.0.chip\n",
      "Rat_AFFY_230_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_230_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_AFFY_RaGene_1_0_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_RaGene_1_0_st_v1_MSigDB.v7.0.chip\n",
      "Rat_AFFY_RaGene_1_0_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_RaGene_1_0_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_AFFY_RaGene_2_1_st_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_RaGene_2_1_st_v1_MSigDB.v7.0.chip\n",
      "Rat_AFFY_RaGene_2_1_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_RaGene_2_1_st_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_AFFY_U34_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_U34_MSigDB.v7.0.chip\n",
      "Rat_AFFY_U34_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_AFFY_U34_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_SurePrint_g3_ge_8x60k_MSigDB.v7.0.chip\n",
      "Rat_Agilent_SurePrint_g3_ge_8x60k_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_SurePrint_g3_ge_8x60k_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_WholeGenome_4x44k_v1_MSigDB.v7.0.chip\n",
      "Rat_Agilent_WholeGenome_4x44k_v1_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_WholeGenome_4x44k_v1_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Agilent_WholeGenome_4x44k_v3_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_WholeGenome_4x44k_v3_MSigDB.v7.0.chip\n",
      "Rat_Agilent_WholeGenome_4x44k_v3_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Agilent_WholeGenome_4x44k_v3_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Codelink_Array_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Codelink_Array_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_ENSEMBL_Gene_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_ENSEMBL_Gene_ID_MSigDB.v7.0.chip\n",
      "Rat_ENSEMBL_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_ENSEMBL_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Gene_Symbol_Remapping_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Gene_Symbol_Remapping_MSigDB.v7.0.chip\n",
      "Rat_Gene_Symbol_Remapping_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Gene_Symbol_Remapping_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_NCBI_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_NCBI_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_Phalanx_OneArray_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Phalanx_OneArray_MSigDB.v7.0.chip\n",
      "Rat_Phalanx_OneArray_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_Phalanx_OneArray_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_RefSeq_ID_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_RefSeq_ID_MSigDB.v7.0.chip\n",
      "Rat_RefSeq_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_RefSeq_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_UniProt_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_UniProt_Gene_ID_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_UniProt_SwissProt_MSigDB.v7.0.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_UniProt_SwissProt_MSigDB.v7.0.chip\n",
      "Rat_UniProt_SwissProt_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_UniProt_SwissProt_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "Rat_UniProt_TrEMBL_to_Human_Orthologs_MSigDB.v7.1.chip = ftp://ftp.broadinstitute.org/pub/gsea/annotations_versioned/Rat_UniProt_TrEMBL_to_Human_Orthologs_MSigDB.v7.1.chip\n",
      "\n",
      "\n",
      "scoring.scheme\n",
      "classic Kolmogorov-Smirnov = classic\n",
      "weighted = weighted\n",
      "weighted_p2 = weighted_p2\n",
      "weighted_p1.5 = weighted_p1.5\n",
      "weighted\n",
      "\n",
      "collapsing.mode.for.probe.sets.with.more.than.one.match\n",
      "Max_probe = Max_probe\n",
      "Median_of_probes = Median_of_probes\n",
      "Mean_of_probes = Mean_of_probes\n",
      "Sum_probes = Sum_probes\n",
      "Max_probe\n",
      "\n",
      "normalization.mode\n",
      "meandiv = meandiv\n",
      "None = None\n",
      "meandiv\n",
      "\n",
      "omit.features.with.no.symbol.match\n",
      "false = false\n",
      "true = true\n",
      "true\n",
      "\n",
      "make.detailed.gene.set.report\n",
      "false = false\n",
      "true = true\n",
      "true\n",
      "\n",
      "create.svgs\n",
      "false = false\n",
      "true = true\n",
      "false\n",
      "\n",
      "create.zip\n",
      "false = false\n",
      "true = true\n",
      "true\n",
      "\n",
      "dev.mode\n",
      "false = false\n",
      "true = true\n",
      "false\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Loop through each parameter\n",
    "for param in params_list:\n",
    "    if param.is_choice_param():        # If the parameter is a choice param\n",
    "        print( param.get_name() )      # Print the parameter's name\n",
    "        \n",
    "        choices = param.get_choices()  # Get a list of valid choices \n",
    "        for choice in choices:         # Print the label and value for each choice\n",
    "            print( choice['label'] + \" = \" + choice['value'] )\n",
    "            \n",
    "        # Print the default selected value for each choice\n",
    "        print( param.get_choice_selected_value() )\n",
    "        print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all\n",
      "7\n",
      "just_hallmarks\n",
      "1\n",
      "\n",
      "2\n",
      "('all', ['ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c1.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c3.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c6.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c7.all.v7.1.symbols.gmt', 'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/h.all.v7.1.symbols.gmt'])\n",
      "\n",
      "('just_hallmarks', ['ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/h.all.v7.1.symbols.gmt'])\n"
     ]
    }
   ],
   "source": [
    "genesets_param = [x for x in params_list if x.get_name().startswith(\"gene.sets.database\")]\n",
    "assert len(genesets_param) == 1, len(genesets_param)\n",
    "genesets_param = genesets_param[0]\n",
    "\n",
    "reference_geneset_urls = []\n",
    "for group_name, group_geneset_labels in reference_genesets:\n",
    "    print(group_name)\n",
    "    \n",
    "    matching_geneset_choices = [x for x in genesets_param.get_choices() if x[\"label\"] in group_geneset_labels]\n",
    "    print(len(matching_geneset_choices))\n",
    "    check = set([x[\"label\"] for x in matching_geneset_choices])\n",
    "    assert check == group_geneset_labels, (len(check), len(group_geneset_labels), check, group_geneset_labels)\n",
    "    \n",
    "    reference_geneset_urls.append((\n",
    "        group_name, [x[\"value\"] for x in matching_geneset_choices]\n",
    "    ))\n",
    "\n",
    "print()\n",
    "print(len(reference_geneset_urls))\n",
    "print(\"\\n\\n\".join([str(x) for x in reference_geneset_urls]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "16\n",
      "[('all', <gp.core.GPJobSpec object at 0x7fcbf2f82410>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf2f82390>), ('all', <gp.core.GPJobSpec object at 0x7fcbf2f823d0>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf2f822d0>), ('all', <gp.core.GPJobSpec object at 0x7fcbf2f82310>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf2f82290>), ('all', <gp.core.GPJobSpec object at 0x7fcbf2f82190>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf2f82790>), ('all', <gp.core.GPJobSpec object at 0x7fcbf2f82810>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf2f82250>), ('all', <gp.core.GPJobSpec object at 0x7fcbf4551210>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf4551290>), ('all', <gp.core.GPJobSpec object at 0x7fcbf4551410>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf4551350>), ('all', <gp.core.GPJobSpec object at 0x7fcbf4551550>), ('just_hallmarks', <gp.core.GPJobSpec object at 0x7fcbf4551450>)]\n"
     ]
    }
   ],
   "source": [
    "all_job_spec_list = []\n",
    "\n",
    "change_default_params = {\"number.of.permutations\": str(num_permutations),\n",
    "                        \"collapse.dataset\": \"No_Collapse\"}\n",
    "\n",
    "add_params = {\"job.memory\":job_memory}\n",
    "    \n",
    "for input_gp_file in input_gp_files_list:\n",
    "    for gs_group_name, gs_urls in reference_geneset_urls:\n",
    "\n",
    "        # Create the GPJobSpec\n",
    "        job_spec = gsea_preranked_module.make_job_spec()\n",
    "#         print(dir(job_spec))\n",
    "\n",
    "        for param_name,param_value in add_params.items():\n",
    "            job_spec.set_parameter(param_name, param_value)\n",
    "\n",
    "        # Loop through all the parameters and set their default values\n",
    "        for param in gsea_preranked_module.get_parameters():  \n",
    "            # If the parameter has a default value, set that value\n",
    "            param_name = param.get_name()\n",
    "            param_default_value = param.get_default_value()\n",
    "#             print(param_name, param_default_value)\n",
    "\n",
    "            if param_name in change_default_params:\n",
    "                job_spec.set_parameter(param_name, change_default_params[param_name])\n",
    "            elif param.get_default_value() != None: \n",
    "                # Set the default value\n",
    "                job_spec.set_parameter(param_name, param_default_value)\n",
    "\n",
    "        # Attach the input file to the correct parameter\n",
    "        job_spec.set_parameter(\"ranked.list\", input_gp_file.get_url()) \n",
    "\n",
    "        job_spec.set_parameter(\"gene.sets.database\", gs_urls)\n",
    "            #[\"ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c1.all.v7.1.symbols.gmt\"])\n",
    "\n",
    "        # job_spec.set_parameter(\"number.of.permutations\", \"10\")\n",
    "#         job_spec.params\n",
    "\n",
    "        all_job_spec_list.append((gs_group_name, job_spec))\n",
    "\n",
    "print(len(all_job_spec_list))\n",
    "print([x for x in all_job_spec_list])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[{'name': 'job.memory', 'values': ['4 Gb']},\n",
       " {'name': 'number.of.permutations', 'values': ['1000']},\n",
       " {'name': 'collapse.dataset', 'values': ['No_Collapse']},\n",
       " {'name': 'scoring.scheme', 'values': ['weighted']},\n",
       " {'name': 'max.gene.set.size', 'values': ['500']},\n",
       " {'name': 'min.gene.set.size', 'values': ['15']},\n",
       " {'name': 'collapsing.mode.for.probe.sets.with.more.than.one.match',\n",
       "  'values': ['Max_probe']},\n",
       " {'name': 'normalization.mode', 'values': ['meandiv']},\n",
       " {'name': 'omit.features.with.no.symbol.match', 'values': ['true']},\n",
       " {'name': 'make.detailed.gene.set.report', 'values': ['true']},\n",
       " {'name': 'num.top.sets', 'values': ['20']},\n",
       " {'name': 'random.seed', 'values': ['timestamp']},\n",
       " {'name': 'create.svgs', 'values': ['false']},\n",
       " {'name': 'output.file.name', 'values': ['<ranked.list_basename>.zip']},\n",
       " {'name': 'create.zip', 'values': ['true']},\n",
       " {'name': 'dev.mode', 'values': ['false']},\n",
       " {'name': 'ranked.list',\n",
       "  'values': ['https://cloud.genepattern.org/gp/users/dllahr/tmp/run5687728369687963663.tmp/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.rnk']},\n",
       " {'name': 'gene.sets.database',\n",
       "  'values': ['ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c1.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c2.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c3.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c5.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c6.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/c7.all.v7.1.symbols.gmt',\n",
       "   'ftp://gpftp.broadinstitute.org/module_support_files/msigdb/gmt/h.all.v7.1.symbols.gmt']}]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(all_job_spec_list[0][0])\n",
    "t = all_job_spec_list[0][1]\n",
    "dir(t)\n",
    "t.params"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "Pending in the GenePattern queue, it has not been submitted to an external queuing system\n",
      "False\n",
      "waiting on job index i:   0\n",
      "waiting on job index i:   1\n",
      "waiting on job index i:   2\n",
      "waiting on job index i:   3\n",
      "waiting on job index i:   4\n",
      "waiting on job index i:   5\n",
      "waiting on job index i:   6\n",
      "waiting on job index i:   7\n",
      "waiting on job index i:   8\n",
      "waiting on job index i:   9\n",
      "waiting on job index i:   10\n",
      "waiting on job index i:   11\n",
      "waiting on job index i:   12\n",
      "waiting on job index i:   13\n",
      "waiting on job index i:   14\n",
      "waiting on job index i:   15\n"
     ]
    }
   ],
   "source": [
    "job_list = []\n",
    "\n",
    "for gs_group_name, job_spec in all_job_spec_list:\n",
    "    # This will return the job object and continue execution even if the job isn't finished\n",
    "    job = gpserver.run_job(job_spec, False)\n",
    "#     print(dir(job))\n",
    "\n",
    "    # Prints a brief description of the job's current state\n",
    "    print( job.get_status_message() )\n",
    "\n",
    "    # Quaries the server and returns True if the job is complete, False otherwise\n",
    "    print( job.is_finished() )\n",
    "\n",
    "    job_list.append((gs_group_name, job))\n",
    "#     job.wait_until_done()\n",
    "\n",
    "for i, (gs_group_name, job) in enumerate(job_list):\n",
    "    print(\"waiting on job index i:  \", i)\n",
    "    job.wait_until_done()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "zip_dir = os.path.join(gsea_dir, \"zip_files\")\n",
    "if not os.path.exists(zip_dir):\n",
    "    os.mkdir(zip_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "https://cloud.genepattern.org/gp/jobResults/257812/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257813/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257814/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257815/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257816/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257817/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257818/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257819/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257820/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257821/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257822/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257823/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_just_hallmarks.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257824/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_all.zip\n",
      "https://cloud.genepattern.org/gp/jobResults/257825/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "********************* ALERT! ************************\n",
      "Zip file not found may mean job failed to run\n",
      "gs_group_name: all\n",
      "*****************************************************\n",
      "https://cloud.genepattern.org/gp/jobResults/257827/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2_just_hallmarks.zip\n",
      "\n",
      "#####################################\n",
      "info about zip_files_list:\n",
      "15\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_all.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_just_hallmarks.zip\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/zip_files/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2_just_hallmarks.zip\n",
      "\n"
     ]
    }
   ],
   "source": [
    "zip_files_list = []\n",
    "\n",
    "no_zip_files = []\n",
    "\n",
    "for gs_group_name, job in job_list:\n",
    "#     print(gs_group_name)\n",
    "\n",
    "    t = [x for x in job.get_output_files() if x.get_name().endswith(\".zip\")]\n",
    "    \n",
    "    if len(t) == 1:\n",
    "        zip_output_file = t[0]\n",
    "    #     print(zip_output_file.get_name())\n",
    "    #     print(zip_output_file.get_url())\n",
    "\n",
    "        dl_filename = os.path.splitext(zip_output_file.get_name())[0] + \"_\" + gs_group_name + \".zip\"\n",
    "    #     print(dl_filename)\n",
    "\n",
    "        dl_filepath = os.path.join(zip_dir, dl_filename)\n",
    "        zip_files_list.append(dl_filepath)\n",
    "\n",
    "        print(zip_output_file.get_url())\n",
    "        print(dl_filepath)\n",
    "\n",
    "        f = open(dl_filepath, \"wb\")\n",
    "        f.write(zip_output_file.open().read())\n",
    "        f.close()\n",
    "    elif len(t) == 0:\n",
    "        print(\"********************* ALERT! ************************\")\n",
    "        print(\"Zip file not found may mean job failed to run\")\n",
    "        print(\"gs_group_name:\", gs_group_name)\n",
    "        no_zip_files.append((gs_group_name, job))\n",
    "        print(\"*****************************************************\")\n",
    "    else:\n",
    "        print(\"********************* WARNING ************************\")\n",
    "        print(\"More than 1 zip file found in job output, skipping\")\n",
    "        print(\"gs_group_name:\", gs_group_name)\n",
    "        print(\"*****************************************************\")\n",
    "    \n",
    "print()\n",
    "print(\"#####################################\")\n",
    "print(\"info about zip_files_list:\")\n",
    "print(len(zip_files_list))\n",
    "print(\"\\n\".join(zip_files_list))\n",
    "print()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "all\n",
      "257826\n",
      "https://cloud.genepattern.org/gp/pages/index.jsf?jobid=257826\n",
      "['stderr.txt', 'stdout.txt']\n"
     ]
    }
   ],
   "source": [
    "for gs_group_name, job in no_zip_files:\n",
    "    print(gs_group_name)\n",
    "    print(job.job_number)\n",
    "    print(job.get_job_status_url())\n",
    "    print([x.get_name() for x in job.get_output_files()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "# dir(no_zip_files[0][1])\n",
    "# no_zip_files[0][1].has"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "# t = no_zip_files[0][1]\n",
    "# t.get_output_files()[0].get_name()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_logFC_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx1_24h_Vehicle_921_QDx1_24h_t_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_logFC_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_921_QDx6_24h_Vehicle_921_QDx6_24h_t_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_logFC_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx1_24h_Vehicle_MP46_QDx1_24h_t_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_all\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_logFC_r30144x2_just_hallmarks\n",
      "/Users/dlahr/temp/H202SC20040591_dge/analysis/gsea/H202SC20040591_FHT3794_MP46_QDx6_24h_Vehicle_MP46_QDx6_24h_t_r30144x2_just_hallmarks\n"
     ]
    }
   ],
   "source": [
    "for zip_file in zip_files_list:\n",
    "#     print(zip_file)\n",
    "    \n",
    "    dir_name = os.path.splitext(\n",
    "        os.path.basename(zip_file)\n",
    "    )[0]\n",
    "    \n",
    "    extract_dir = os.path.join(gsea_dir, dir_name)\n",
    "    print(extract_dir)\n",
    "    \n",
    "    if os.path.exists(extract_dir):\n",
    "        shutil.rmtree(extract_dir)\n",
    "    os.mkdir(extract_dir)\n",
    "    \n",
    "    my_zipfile = zipfile.ZipFile(zip_file, \"r\")\n",
    "    my_zipfile.extractall(extract_dir)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
